web_generator/lib/src/ast/builtin.dart

```diff
@@ -95,6 +95,37 @@ class BuiltinType extends Type {
           isNullable: isNullable),
     };
   }
+
+  static BuiltinType? referred(String name,
+      {bool? isNullable, List<Type> typeParams = const []}) {
+    final jsName = switch (name) {
+      'Array' => 'JSArray',
+      'Promise' => 'JSPromise',
+      'ArrayBuffer' => 'JSArrayBuffer',
+      'Function' => 'JSFunction',
+      'DataView' => 'JSDataView',
+      'Float32Array' => 'JSFloat32Array',
+      'Float64Array' => 'JSFloat64Array',
+      'Int8Array' => 'JSInt8Array',
+      'Int16Array' => 'JSInt16Array',
+      'Int32Array' => 'JSInt32Array',
+      'Int64Array' => 'JSInt64Array',
+      'Uint8Array' => 'JSUint8Array',
+      'Uint16Array' => 'JSUint16Array',
+      'Uint32Array' => 'JSUint32Array',
+      'Uint8ClampedArray' => 'JSUint8ClampedArray',
+      _ => null
+    };
+    final jsTypeArgs = switch (name) { 'Array' || 'Promise' => 1, _ => 0 };
```

Comment:
Nothing to do in this PR, but it would be nice if we can compute this rather than assume it. It would make making things like `JSFunction` generic less breaking.